generated/provider_dependencies.json

```diff
@@ -404,6 +404,7 @@
     "devel-deps": [],
     "plugins": [],
     "cross-providers-deps": [
+      "common.compat",
```

Comment:
Not really sure why, but this was added: https://github.com/apache/airflow/actions/runs/10038905531/job/27741833243

Reply:
That's automated detection of cross-provider dependencies.

It's added automatically when one provider imports another (which is the case here). 

But in your case you also need to add `common.compat>=1.1.0" as explicit dependency of the provider when you are using features from upcoming 1.1.

Reply:
@potiuk not that I disagree with that mechanism - I think capturing implicit dependencies is fine - but should it have explicit dependency when it only imports stuff behind `TYPE_CHECKING`?

Reply:
I believe so. Compat must be installed in order to use the new method. Maybe we could add some feature to mark it as optional if you think it makes sense in this case, but I am not sure if it's really optional. I am not sure if just TYPE_CHECKING is enough - generally the assumption was that if you use or expect anything coming from different providers, it is added as cross-dependency. For example it is also used to see if you should run tests for the other provider if the "cross-related" one has been modified (and in this case it's good because it mght cause problems and failing tests if it did not.

FIY the code doing it is here:

https://github.com/apache/airflow/blob/e30f8102b2dfc2c99454c99c286138754e4a1f1c/scripts/ci/pre_commit/update_providers_dependencies.py#L167

So if you think we can improve that ... :)

Reply:
BTW. It's the first time it happened, generally I think providers are using other providers more "extensively".